### PR TITLE
aws - tags - copy-related-tag using resourcegroupstaggingapi, support tags as key

### PR DESCRIPTION
**Pull request by thisisshi**
closes: https://github.com/cloud-custodian/cloud-custodian/issues/7188

Sample Policy:

```yaml
policies:
  - name: copy-from-instance-profile
    resource: aws.ec2
    actions:
      - type: copy-related-tag
        resource: resourcegroupstaggingapi
        skip_missing: true
        key: "IamInstanceProfile.Arn"
        tags: '*'
  - name: copy-from-tag
    resource: aws.ec2
    actions:
      - type: copy-related-tag
        resource: resourcegroupstaggingapi
        skip_missing: true
        key: "tag:test"
        tags: '*'
```

todo:
- [x] tests
- [x] docs

### CODE REVIEW

File: c7n/tags.py [hunk 1]
```diff
@@ -1032,15 +1054,26 @@ def validate(self):
 
     def process(self, resources):
         related_resources = []
-        for rrid, r in zip(jmespath.search('[].[%s]' % self.data['key'], resources),
+        search_path = "[].[%s]" % self.data['key']
```
Comment (ajkerrigan) [inline c7n/tags.py:1057]: nit: looks like this could go in an else below, since it'll be redefined for the `tag:` case

File: c7n/tags.py [hunk 2]
```diff
@@ -1101,6 +1134,26 @@ def get_resource_tag_map(self, r_type, ids):
             for r in manager.get_resources(list(ids))
         }
 
+    def get_resource_tag_map_universal(self, ids):
+        related_region = None
+        ids = list(ids)
+        if ids:
+            if ids[0].startswith('arn:aws:iam'):
```
Comment (ajkerrigan) [inline c7n/tags.py:1141]: this seems like a reasonable way to work around the fact that we can't check for the resource type's `global_resource` property. i see a few other global resource types (account, support, shield, route53), but as far as i can tell none of them are relevant for this action. guessing you went through the same review :) :+1:

**Review by ajkerrigan — APPROVED** [review]
looks good, thanks! :+1:

File: c7n/tags.py [hunk 3]
```diff
@@ -998,6 +998,25 @@ class CopyRelatedResourceTag(Tag):
...
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+                - name: copy-tags-from-unsupported-resource
+                  resource: ebs-snapshot
+                  actions:
+                    - type: copy-related-tag
```
Comment (kapilt) [inline c7n/tags.py:1015]: is this example complete?, it doesn't logically follow where the tags are being copied from.

Comment (thisisshi) [reply c7n/tags.py:1015]: Yes, the resource that the tags would be pulled from is the arn found in the tag

**Pull request merged**